core/dbt/events/format.py

```diff
@@ -3,6 +3,22 @@
 from typing import Optional, Union
 
 
+node_states = {
+    "queued": "queued",
+    "skipped": "skipped",
+    "started": "started",
+    "success": "success",
+    "failed": "failed"
+}
+
+node_statuses = {
+    "running": "running",
+    "pass": "pass",
+    "error": "error",
+    "fail": "fail"
+}
```

Comment:
- I'm thinking this should be pulled off the [result contracts](https://github.com/dbt-labs/dbt-core/blob/7f2d3cd24f8715d3e36873a1f0049d8a764b2df5/core/dbt/contracts/results.py#L61-L89), rather than re-defined here
- Can we consolidate these into one `status`? I'm thinking we don't need separate `state` + `status`, since every node's progression should look like:
    - `started`
    - `compiling` (if applicable)
    - `executing` / `running` (neither word is perfect, both work)
    - final result status (one of the contracted options)

Reply:
This sounds good to me.

Reply:
I agree as I wrote this.  However, I think i named these poorly.  I want to make sure I'm not missing the original intention.   A `node_state` and a `run_status` were what was originally laid out as needed.  `node_state` being state of the specific node processing (`started`, `compiling`, `executing`, `skipped`, `failed`, `success`) and a separate `run-status` (`running`, `pass`, `error`, `fail`)

Reply:
I'm not sure here. When Jeremy comes back online I think he'll know more.

Reply:
I think the middle two (`compiling` + `executing`) imply the existence of new events that we'll need to add [here](https://github.com/dbt-labs/dbt-core/blob/7f2d3cd24f8715d3e36873a1f0049d8a764b2df5/core/dbt/task/base.py#L279-L297)

Reply:
Ah, sorry, I missed the question above earlier on. My read of `run_state`:
- This is the name of the field in older structured logging. It's only ever `running` or `internal`, neither of which is very helpful.
- I don't much like the name. If we _were_ to have a field named `run_state`, that should mean the state of the overall `run`: task level, all nodes, not specific to this particular node. This isn't a very interesting field — I think it's fair to assume that, if we're firing an event that reports on the status of a specific node, the overall run state has got to be `running`. Of course, we should + do fire other events when the run concludes, including its final status.
- At first, it feels like we need different fields for (a) the state of the currently running node, and (b) the final status of the node once it finishes running. Given that these are effectively non-overlapping categories, I think we can easily combine them into a single `node_status`. An alternative approach would be to have a `current_node_state` that reports `finished` once it has finished and the `node_status` is available. I'm inclined to prefer the sparser option, and I don't think we're overloading its meaning by having it represent the status of the node both while running and once finished.

@drewbanin feel free to weigh in if you disagree with any of the above!

Reply:
oh right I ran into `running` vs `internal` in our tests so we talked about this a bit. I like the sound of merging these into a single field.